### PR TITLE
Restructure of code to use modern syntax such as async/await

### PR DESCRIPTION
**Pull request by plusk01**
This PR is essentially a complete overhaul of the code, but it contains the same behavior and functionality. I replaced the deep nested promises with `async` and `await` to enhance readability. Additionally, I made functions smaller and eliminated some global variables.

### CODE REVIEW

**Comment by spruce** [comment]
Gonna review it in the next few days. Thanks for commiting this and pouring so much effort into it :)

File: index.js [hunk 1]
```diff
@@ -1,500 +1,589 @@
...
+
+        let bodyConverted = convertIssuesAndComments(note.body, note);
+
+        // process asynchronous code in sequence -- treats kind of like blocking
+        await github.issues.createComment({
+                    owner: settings.github.owner,
+                    repo: settings.github.repo,
+                    number: ghIssue.number,
+                    body: bodyConverted
+                  }).catch(x=>{});
```
Comment (spruce) [inline index.js]: I know the original did not care about it but isn't it better at least console.log or maybe even exit program?

File: index.js [hunk 2]
```diff
@@ -1,500 +1,589 @@
...
+
+  // if a GitLab milestone does not exist in GitHub repo, create it.
+  for (let milestone of milestones) {
+      if (!ghMilestones.find(m => m.title === milestone.title)) {
+        console.log("Creating: " + milestone.title);
+        try {
+
+          // process asynchronous code in sequence
+          await (() => {
+            createMilestone(settings.github.owner, settings.github.repo, milestone);
```
[the remark this answers is not in the document]
Comment (plusk01) [reply index.js]: You're right, it is not. This was leftover from some old code.

File: index.js [hunk 3]
```diff
@@ -1,500 +1,589 @@
...
-      var l = ghIssues.data.length;
-      for (var i = 0; i < l; i++) {
-        allGhIssues[allGhIssues.length] = ghIssues.data[i];
+
+      if (!ghLabels.find(l => l === label.name)) {
+        console.log("Creating: " + label.name);
+        try {
+
+          // process asynchronous code in sequence
+          await (() => {
```
[the remark this answers is not in the document]
Comment (plusk01) [reply index.js]: agreed

File: index.js [hunk 4]
```diff
@@ -1,500 +1,589 @@
...
+  // Get a list of all milestones associated with this project
+  let milestones = await gitlab.ProjectMilestones.all(projectId);
+
+  // sort milestones in ascending order of when they were created (by id)
+  milestones = milestones.sort((a, b) => a.id - b.id);
+
+  // get a list of the current milestones in the new GitHub repo (likely to be empty)
+  let ghMilestones = await getAllGHMilestones(settings.github.owner, settings.github.repo);
+
+  inform("Transferring Milestones");
```
[the remark this answers is not in the document]
Comment (plusk01) [reply index.js]: done

File: index.js [hunk 5]
```diff
@@ -1,500 +1,589 @@
...
+ * Transfer any labels that exist in GitLab that do not exist in GitHub.
+ */
+async function transferLabels(projectId, attachmentLabel = true, useLowerCase = true) {
+  // Get a list of all labels associated with this project
+  let labels = await gitlab.Labels.all(projectId);
+
+  // get a list of the current label names in the new GitHub repo (likely to be just the defaults)
+  let ghLabels = await getAllGHLabelNames(settings.github.owner, settings.github.repo);
+
+  inform("Transferring Labels")
```
[the remark this answers is not in the document]
Comment (plusk01) [reply index.js]: done

File: index.js [hunk 6]
```diff
@@ -1,500 +1,589 @@
...
+  // TODO return all issues via pagination
+  let issues = await gitlab.Issues.all({projectId: projectId});
+
+  // sort issues in ascending order of when they were created (by id)
+  issues = issues.sort((a, b) => a.id - b.id);
+
+  // get a list of the current issues in the new GitHub repo (likely to be empty)
+  let ghIssues = await getAllGHIssues(settings.github.owner, settings.github.repo);
+
+  inform("Transferring " + issues.length.toString() + " Issues");
```
[the remark this answers is not in the document]
Comment (plusk01) [reply index.js]: done

File: index.js (hunk 1 above)
Comment (plusk01) [reply index.js]: good point :-)

File: index.js [hunk 7]
```diff
@@ -1,500 +1,589 @@
...
+/**
+ * Transfer any issues and their comments that exist in GitLab that do not exist in GitHub.
+ */
+async function transferIssues(owner, repo, projectId) {
+
+  // Because each 
+  let milestoneData = await getAllGHMilestones(owner, repo);
+
+  // get a list of all GitLab issues associated with this project
+  // TODO return all issues via pagination
```
[the remark this answers is not in the document]
Comment (plusk01) [reply index.js]: I didn't create a helper here for consistency -- only the GitHub data getters have helper functions.

**Comment by plusk01** [comment]
Alright @spruce, I've just pushed those fixes for you. Sorry for the delay, I've just recently moved!

File: index.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (spruce) [reply index.js]: ok

**Comment by spruce** [comment]
Thanks for making the effort to create this PR. No worries about the delay life get's into the way more often than not for me, too.

**Pull request merged**